### ISSUE TITLE
Memory issues

### ISSUE THREAD

**Issue by DavidGOrtega** [FL33TW00D/whisper-turbo#38]
Experimenting with the project I found two issues: 

 1. GPU memory is not released after the infer. 
 2. The bigger the audio file the bigger the memory consumed. For a 4min file it drains on my computer 4,48GB. Bigger files has frozen my computer draining up to 20GB. 

Macos M1 16GB Ventura 13.1
Chromium 118.0.5993.32

**Comment by FL33TW00D** [comment]
I’m aware of the leak! Fix coming during the next big release.

I’m on vacation at the moment, fix coming in November

**Comment by DavidGOrtega** [comment]
I have tried to free the session after the stream or run

```js
const result = Result.ok(await this.whisperSession.stream(audio, callback)) as Result<void, Error>;
this.whisperSession.free();
return result;
```

but did not work, maye be it has to be done at a segment level?

However despite that Segment is meant to be 

```js
export class Segment {
  free(): void;
}
```

It returns another kind of object.

Are you using [laserbreak](https://github.com/FL33TW00D/laserbeak) in `whisper-gpu`?

**Comment by DavidGOrtega** [comment]
> I’m on vacation at the moment, fix coming in November

Happy holidays! 🌴

**Comment by FL33TW00D** [comment]
> I have tried to free the session after the stream or run
> 
> ```js
> const result = Result.ok(await this.whisperSession.stream(audio, callback)) as Result<void, Error>;
> this.whisperSession.free();
> return result;
> ```
> 
> but did not work, maye be it has to be done at a segment level?
> 
> However despite that Segment is meant to be
> 
> ```js
> export class Segment {
>   free(): void;
> }
> ```
> 
> It returns another kind of object.
> 
> Are you using [laserbreak](https://github.com/FL33TW00D/laserbeak) in `whisper-gpu`?

This won’t work unfortunately- the problem is deeper. 

Laserbeak isnt being used in this project.

**Comment by FL33TW00D** [comment]
> > I’m on vacation at the moment, fix coming in November
> 
> Happy holidays! 🌴

@DavidGOrtega Leak is fixed, will be released in 0.8.0! Thanks for letting me know and testing it out!
